src/org/zaproxy/zap/extension/ascanrulesBeta/HPP.java

```diff
@@ -154,7 +157,12 @@ public void scan() {
 					log.debug("Injecting payload...");
 					HttpMessage newMsg = getNewMsg();
 					newMsg.setGetParams(tags);
-					sendAndReceive(newMsg);
+					try {
+						sendAndReceive(newMsg);
+					} catch (IllegalStateException|UnknownHostException ex) {
+						log.info("Caught " + ex.getClass().getName() + " when accessing: " + newMsg.getRequestHeader().getURI().toString() + 
+								"\n The target may have replied with a poorly formed redirect due to our input.");
```

Comment:
In this case the test should not continue, the response would not be from the "attack" request?